protocol/x/listing/types/msg_upgrade_isolated_perpetual_to_cross.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	fmt "fmt"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgUpgradeIsolatedPerpetualToCross{}
+
+func (msg *MsgUpgradeIsolatedPerpetualToCross) GetSigners() []sdk.AccAddress {
+	addr, _ := sdk.AccAddressFromBech32(msg.Authority)
+	return []sdk.AccAddress{addr}
+}
+
+func (msg *MsgUpgradeIsolatedPerpetualToCross) ValidateBasic() error {
```

Comment:
Why is this needed? I don't see it being used in the msg execution

Reply:
Other messages use this pattern for basic / stateless validation: https://github.com/dydxprotocol/v4-chain/blob/76c18b66f07d5f649869929fa9edfeb846f10cc8/protocol/x/prices/types/message_create_oracle_market.go

In those cases it is also not called in the msg execution, but as part of test suites. Not sure if something upstream calls the validation during execution or not. But either way this matches that pattern

Reply:
ValidateBasic has been deprecated
![image](https://github.com/user-attachments/assets/e46f4b58-8ebb-43f1-9904-35fe2ca0642d)